app/features/quizzify/tools.py

```diff
@@ -183,21 +114,56 @@ def load(self, tool_files: List[ToolFile]) -> List[Document]:
 
         # Pass Queue to the file loader if there are any successful loads
         if any_success:
-            file_loader = self.loader(queued_files)
+            file_loader = self.loader(queued_files, files_list)
             documents = file_loader.load()
 
             if self.verbose:
                 logger.info(f"Loaded {len(documents)} documents")
 
-        if not any_success:
-            raise LoaderError("Unable to load any files from URLs")
 
         return documents
 
+class BaseLoader:
+
+    def __init__(self, verbose = False):
+        self.verbose = verbose
+
+    def load(self, files):
+        
+        documents = []
+        file_type_list = ['csv','pptx','docx','pdf','txt']
+        youtube_files = []
+        webpage_files = []
+        other_files = []
+
+        for file in files:
+            url = file.url
+            file_type = file.filetype
+            
+            if url.lower().startswith('https://www.youtube.com'):
+                youtube_files.append(file)
+            elif file_type in file_type_list:
+                other_files.append(file)
+            elif url.lower().startswith('http://') or url.lower().startswith('https://'):
+                webpage_files.append(file)
+            else:
+                raise ValueError(f"Received {file_type}, Unsupported File Type\n"
+                                    f"Supported File Types 'pdf, txt','web page url','pptx','csv','docx','Youtube Url'")
+
+
+        docs = CustomYoutubeLoader(verbose = self.verbose).load(youtube_files)
+        documents.extend(docs)
```

Comment:
This flow can be complex to accomplish if there is only one file type. Refactor this for a better workflow.